test/unit/parser/dvcondition_location.d

```diff
@@ -0,0 +1,84 @@
+module parser.dvcondition_location;
+
+import dmd.frontend : parseModule;
+import support : afterEach, beforeEach;
+import dmd.cond : VersionCondition, DebugCondition;
+import dmd.globals : Loc;
+import dmd.visitor : SemanticTimeTransitiveVisitor;
+
+@beforeEach initializeFrontend()
+{
+    import dmd.frontend : initDMD;
+
+    initDMD();
+}
+
+@afterEach deinitializeFrontend()
+{
+    import dmd.frontend : deinitializeDMD;
+    deinitializeDMD();
+}
+
+extern (C++) static class Visitor : SemanticTimeTransitiveVisitor
+{
+    alias visit = typeof(super).visit;
+    Loc l;
+
+    override void visit(VersionCondition vc)
+    {
+        l = vc.loc;
+    }
+
+    override void visit(DebugCondition dc)
+    {
+        l = dc.loc;
+    }
+}
+
+@("version(identifier)")
+unittest
+{
+    auto t = parseModule("test.d", "version(a)");
+
+    scope visitor = new Visitor;
+    t.module_.accept(visitor);
+
+    assert(visitor.l.linnum == 1);
+    assert(visitor.l.charnum == 9);
```

Comment:
So linnum and charnum are that of the condition ? What happens if I write:
```D
version (
Hello
)
```
Likewise, looking at the code, this is inconsistent with the way plain `debug` are treated, where the linnum / charnum will be on the start of the `debug` token.

Reply:
(Aside: at some point, we should really have location ranges in all nodes, so that the start and end locations are recorded).

Reply:
> So linnum and charnum are that of the condition

I agree. The location should point to the first byte of `version`.

Reply:
> (Aside: at some point, we should really have location ranges in all nodes, so that the start and end locations are recorded).

Yes, that's what I've already suggested. It looks like @AsterMiha is working in that direction based on my feedback: https://github.com/dlang/dmd/pull/11788#issuecomment-698186023.

Reply:
> > So linnum and charnum are that of the condition
> 
> I agree. The location should point to the first byte of `version`.

I was thinking of adding the location of `version` and `debug` to the ConditionalDeclaration nodes as these are the ones that would contain these conditions